arangod/Aql/AqlFunctionFeature.cpp

```diff
@@ -476,6 +474,11 @@ void AqlFunctionFeature::addMiscFunctions() {
   add({"WITHIN", ".h,.,.,.|.", Function::makeFlags(FF::Cacheable), &Functions::NotImplemented});
   add({"WITHIN_RECTANGLE", "h.,.,.,.,.", Function::makeFlags(FF::Cacheable), &Functions::NotImplemented});
   add({"FULLTEXT", ".h,.,.|.", Function::makeFlags(FF::Cacheable), &Functions::NotImplemented});
+  
+  // this is an internal function that is only here for testing. it cannot
+  // be invoked by end users, because refering to internal functions from user
+  // queries will pretend these functions do not exist.
+  add({"INTERNAL", "", Function::makeFlags(FF::Internal), &Functions::NotImplemented});
```

Comment:
```suggestion
#ifdef ARANGODB_USE_GOOGLE_TESTS
  // this is an internal function that is only here for testing. it cannot
  // be invoked by end users, because refering to internal functions from user
  // queries will pretend these functions do not exist.
  add({"INTERNAL", "", Function::makeFlags(FF::Internal), &Functions::NotImplemented});
#endif
```

Reply:
This is not 100% equivalent.
The `ARANGODB_USE_GOOGLE_TESTS` is defined when the CMake variable `USE_GOOGLE_TESTS` is set. It controls whether gtest C++ tests are compiled or not.
The JavaScript tests however are always available and can be run, regardless of the value of `*_GOOGLE_TESTS`. So it is possible to run them without setting the CMake variable, and in that case, they would fail because they would refer to an unknown function.

Probably nobody will hit this problem in reality, but I would like to keep it as is. The overhead of unconditionally adding the `INTERNAL` function to the list of all functions should be negligible IMHO.

Reply:
Ok, then let's leave it as is.

Reply:
Thanks!